src/views/Player.vue

```diff
@@ -153,11 +164,44 @@ export default {
       return dateTime
     },
 
+    waitFor(conditionFunction) {
+      const poll = resolve => {
+        if(conditionFunction()) resolve();
+        else setTimeout(() => poll(resolve), 400);
+      }
+      return new Promise(poll);
+    },
```

Comment:
This is a promise based utility function that can be reused. You can give it a condititon to check and it will wait for 400ms until that condition is met. I think we can change the 400ms to a globally defined variable